src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WsTrustTokenParameters.cs

```diff
@@ -52,8 +52,11 @@ protected WSTrustTokenParameters(WSTrustTokenParameters other)
                 AdditionalRequestParameters.Add(parameter);
 
             CacheIssuedTokens = other.CacheIssuedTokens;
-            foreach (var claimType in ClaimTypes)
-                ClaimTypes.Add(claimType);
+            Claims = other.Claims;
+            foreach(var element in other.AdditionalRequestParameters)
```

Comment:
You can't just copy the collection over. You have to do a deep copy and copy each item. The reason is that the you don't want changes to the original instance to appear in the copied instance.

Reply:
I see, updated both Claims and AdditionalRequestParameters, could you please help double confirm?